src/templates/base.html

```diff
@@ -31,9 +31,30 @@
     <div id="navigation">
       {% block navigation %}{% endblock %}
     </div>
+    <select class="language-switcher">
+      <option
+        selected="selected"
+        {% block default %}
+        value={{ url_for(request.endpoint, lang=self.lang()[0:2]) }}
```

Comment:
I've looked into [url_for](http://flask.pocoo.org/docs/1.0/api/#flask.url_for) and this is one possible way to generate dynamic URLs.

However, pages below take in more variable arguments:
1. `/<lang>/<year>/outline`
2. `/<lang>/<year>/contributors`
3. `/<lang>/<year>/methodology` 
4. `/<lang>/<year>/<chapter>`

There might also be other new pages that take in more/different variable arguments, thus this block has to be repeatedly customized for the different types of pages. 

Not sure if this is the best approach, what do you think?

Reply:
This looks super weird but I think we can build the arguments dynamically by using the `request.view_args` dict and updating it with the appropriate `lang` value:

```py
value={{ url_for(request.endpoint, **(request.view_args.update(lang=language) or request.view_args)) }}
```

Because the `update` method updates the dict in-place and returns `None` I added the `or request.view_args` to ensure the dict is passed into `url_for`.

Hope that makes sense.

Reply:
Yup that makes sense! Thanks for the explanation 😄